pages/services/data-science-engine/1.0.0/hdfs/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: layout.pug
+navigationTitle: HDFS
+excerpt: Using HDFS with DC/OS Data Science Engine
+title: HDFS
+menuWeight: 4
+model: /services/data-science-engine/data.yml
+render: mustache
+---
+
+If you plan to read and write from HDFS using {{ model.techName }}, there are two Hadoop configuration files that you should include in the {{ model.productName }} classpath:
+
+- `hdfs-site.xml`, which provides default behaviors for the HDFS client.
+- `core-site.xml`, which sets the default file system name.
+
+You can specify the location of these files at install time or for each job.
+
+# Configuring HDFS with {{ model.techName }}
+
+Within the {{ model.techName }} service configuration, set `hdfs.config-url` to be a URL that serves your `hdfs-site.xml` and `core-site.xml`. The following example uses `http://mydomain.com/hdfs-config/hdfs-site.xml` and `http://mydomain.com/hdfs-config/core-site.xml` URLs:
+
+```json
+{
+ "hdfs": {
+   "config-url": "http://mydomain.com/hdfs-config"
+ }
+}
+```
+You can also specify the URLs through the UI. If you are using the default installation of HDFS from Mesosphere, this is probably `http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints`.
+
+## Adding HDFS configuration files per-job
+To add the configuration files manually for a job, use 
+
+```bash
+--conf spark.mesos.uris=<location_of_hdfs-site.xml>,<location_of_core-site.xml>
+```
+
+This setting downloads the files to the sandbox of the driver {{ model.productName }} application, and {{ model.techName }} automatically loads these files into the correct location.
+
+<p class="message--important"><strong>IMPORTANT: </strong>It is important that these files are called <tt>hdfs-site.xml</tt> and <tt>core-site.xml</tt>.</p>
+
+## {{ model.techName }} checkpointing
+To use {{ model.techName }} with checkpointing, make sure you follow the instructions here and use an HDFS directory as the checkpointing directory. For example:
+
+```bash
+val checkpointDirectory = "hdfs://hdfs/checkpoint"
+val ssc = ...
+ssc.checkpoint(checkpointDirectory)
+```
+The HDFS directory is automatically created on HDFS. 
+<!-- The {{ model.techName }} streaming app will work from checkpointed data, even in the event of an application restarts or failure. -->
+
+# S3
+
+You can read/write files to S3 using environment variable-based secrets to pass your AWS credentials.
+
+1. Upload your credentials to the DC/OS secret store:
+
+```bash
+dcos security secrets create <secret_path_for_key_id> -v <AWS_ACCESS_KEY_ID>
+dcos security secrets create <secret_path_for_secret_key> -v <AWS_SECRET_ACCESS_KEY>
+```
+
+1. After uploading your credentials, {{ model.techName }} jobs can get the credentials directly:
+
+```bash
```

Comment:
Actual S3 test example but kind of large: https://github.com/mesosphere/jupyter-service/blob/master/tests/notebooks/tensorflow_on_spark_mnist_gpu_s3.ipynb

Reply:
See https://github.com/mesosphere/dcos-docs-site/pull/2459